core/src/test/java/org/apache/iceberg/hadoop/TestCachingCatalog.java

```diff
@@ -324,6 +324,18 @@ public void testCacheExpirationIsDisabledByANegativeValue() throws IOException {
         catalog.isCacheExpirationEnabled());
   }
 
+  @Test
```

Comment:
Added a test to verify that all the chained  CachingCatalogs don't have the table present in their cache.